cli/src/cli_util.rs

```diff
@@ -1521,8 +1523,8 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
         // `jj new <conflicted commit>` doesn't result in a message about new conflicts.
         let conflicts = RevsetExpression::filter(RevsetFilterPredicate::HasConflict)
             .intersection(&RevsetExpression::filter(RevsetFilterPredicate::File(None)));
-        let removed_conflicts_expr = added_heads.range(&removed_heads).intersection(&conflicts);
-        let added_conflicts_expr = removed_heads.range(&added_heads).intersection(&conflicts);
+        let removed_conflicts_expr = new_heads.range(&removed_heads).intersection(&conflicts);
+        let added_conflicts_expr = old_heads.range(&added_heads).intersection(&conflicts);
```

Comment:
or maybe `new_heads..old_heads` and `old_heads..new_heads`?

It doesn't make sense to filter head commits without considering ancestors.